test/unit/parser/format_parse_test.cpp

```diff
@@ -680,17 +680,43 @@ TEST_F(format_parse_test, subcommand_parser_success)
 
 TEST_F(format_parse_test, subcommand_parser_error)
 {
+    auto parser = get_parser();
+
+    constexpr std::string_view expected_message =
+        "Too many arguments provided. Please see -h/--help for more information.\n"
+        "If you intended to execute a subcommand, it is possible that you have misspelled it.\n"
+        "Available subcommands are: [sub1].\n";
+
+    auto check_error = [&parser, &expected_message](char const * file, int line)
+    {
+        ::testing::ScopedTrace trace(file, line, "check_error");
```

Comment:
The output looks like this:


```
../../../../test/unit/parser/format_parse_test.cpp:701: Failure
Expected equality of these values:
  expected_message
    Which is: "[...]"
  actual_message
    Which is: "[...]"
With diff:
@@ -1,3 +1,2 @@
-[...]
-[...]
+[...]

Google Test trace:
../../../../test/unit/parser/format_parse_test.cpp:719: check_error
```

Without the trace, it would report line `701`, i.e., the `EXPECT_EQ` in the lambda.
Not very helpful.

The trace adds the last part, which then correctly points to line `719`.

```cpp
parser = /* ... */;
{
    SCOPED_TRACE("test1");
    check_error();
}

parser = /* ... */;
{
    SCOPED_TRACE("test2");
    check_error();
}
```

Would also work instead, but too many scopes for my test.

We might unify the `check_error` thing at some point since we use it in multiple locations.